src/vs/workbench/parts/extensions/node/extensionsWorkbenchService.ts

```diff
@@ -799,6 +799,32 @@ export class ExtensionsWorkbenchService implements IExtensionsWorkbenchService {
 			});
 	}
 
+	addToWorkspaceRecommendations(extension: IExtension): TPromise<void> {
+		return this.tipsService.addToWorkspaceRecommendations(extension.id);
+	}
+
+	installAllWorkspaceRecommendations(): TPromise<void> {
+		return this.tipsService.getWorkspaceRecommendations()
+			.then(extensions => {
+				this.queryGallery({ names: extensions })
+					.done(result => {
+						if (result.total < 1) {
+							return;
+						}
+
+						const extension = result.firstPage[0];
+						const promises = [this.open(extension)];
+
+						if (this.local.every(local => local.id !== extension.id)) {
+							promises.push(this.install(extension));
+						}
+
```

Comment:
Not sure if I misunderstood.. I see it is installing only one recommended extensions in the list not all.. I do not see complete result is being read to install?